monitor/plugins/spark_progress.py

```diff
@@ -30,58 +30,108 @@
+
+                # Job Time
+                self.job_expected_time = (self.remaining_time
+                                 / (float(self.number_of_jobs)
+                                 - float(self.current_job_id)))
 
-            progress = result['numCompletedTasks'] / float(result['numTasks'])
-            progress = float("{:10.4f}".format(progress))
-            # Add to metric_info values for this measurement:
-            # value and timestamp
-            ref_value = self._get_elapsed_time(result['submissionTime']) / float(self.expected_time)
-            error = progress - ref_value
-            # The Monasca metric must have the 3 following fields to be created: name, value and timestamp,
-            # but also is possible to increment the metrics identities - using dimensions -  and informations -
-            # using value_meta, a dictionary that contains aditional information about the measurement  .
-            application_progress_error['name'] = 'application-progress.error'
-            application_progress_error['value'] = error
-            application_progress_error['timestamp'] = time.time() * 1000
-            application_progress_error['dimensions'] = self.dimensions
-            # Sending the metric to Monasca
-            self.monasca.send_metrics([application_progress_error])
+            elif current_job['status'] == 'RUNNING':
+                # Job Progress
+                job_progress = (current_job['numCompletedTasks']
+                                / float(current_job['numTasks']))
 
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
 
-            print "Metric successfully published"
+                # Reference Value
+                ref_value = (elapsed_time / self.job_expected_time)
+
+                # Error
+                error = job_progress - ref_value
+                
+                application_progress_error['name'] = ('application-progress'
+                                                      '.error')
+
+                application_progress_error['value'] = error
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+
+                print application_progress_error['value']
+
+                self.monasca.send_metrics([application_progress_error])
 
             time.sleep(MONITORING_INTERVAL)
 
+
+    def _get_elapsed_time(self, gmt_timestamp):
+        try:
+            local_tz = tzlocal.get_localzone()
+
+        except Exception as e:
+            local_tz = "America/Recife"
+            local_tz = pytz.timezone(local_tz)
+
+        date_time = datetime.strptime(gmt_timestamp, '%Y-%m-%dT%H:%M:%S.%fGMT')
+        date_time = date_time.replace(tzinfo=pytz.utc).astimezone(local_tz)
+        date_time = date_time.replace(tzinfo=None)
```

Comment:
I agree with Iury, why so many replaces here?

Reply:
The master was working with these replaces. I just reallocated them. I think this is something necessary to adjust the format of received timestamp. Do you think i should insert all the 'replaces' in just one line? @tellesnobrega

Reply:
No, this is fine.